File: TASK-03/elixir/subtask4.exs
{n, _} = File.read!("input.txt") |> Integer.parse()

File.write!("output.txt",
  Enum.map_join(0..div(n, 2), "\n", fn i ->
    String.duplicate(" ", div(n, 2) - i) <> String.duplicate("*", 2 * i + 1)
  end) <>
  Enum.map_join((div(n, 2) - 1)..0, "\n", fn i ->
    String.duplicate(" ", div(n, 2) - i) <> String.duplicate("*", 2 * i + 1)
  end) <> "\n")

File: TASK-03/rust/subtask4.rs
use std::fs;

fn main() {
    let n: usize = fs::read_to_string("input.txt")
        .expect("Failed to read file")
        .trim()
        .parse()
        .expect("Please enter a valid number");

    let mut output = String::new();
    for i in 0..=n / 2 {
        output.push_str(&format!("{}{}\\n", " ".repeat(n / 2 - i), "*".repeat(2 * i + 1)));
    }
    for i in (0..n / 2).rev() {
        output.push_str(&format!("{}{}\\n", " ".repeat(n / 2 - i), "*".repeat(2 * i + 1)));
    }
    fs::write("output.txt", output).expect("Unable to write file");
}
